source/gui/addonStoreGui/viewModels.py

```diff
@@ -0,0 +1,663 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+import typing
+import enum
+from os import (
+	PathLike,
+)
+from enum import (
+	Enum,
+)
+from typing import (
+	List,
+	Optional,
+	Dict,
+	Tuple,
+)
+import threading
+
+import addonHandler
+import core
+import extensionPoints
+from addonHandler import addonVersionCheck
+from utils.displayString import DisplayStringEnum
+from addonStore.dataManager import DataManager
+from logHandler import log
+
+from addonStore.models import (
+	AddonDetailsModel,
+)
+
+if typing.TYPE_CHECKING:
+	# Remove when https://github.com/python/typing/issues/760 is resolved
+	from _typeshed import SupportsLessThan
+
+
+@enum.unique
+class AvailableAddonStatus(DisplayStringEnum):
+	""" Values to represent the status of add-ons within the NVDA add-on store.
+	Although related, these are independent of the states in L{addonHandler}
+	"""
+	# noinspection PyArgumentList
+	AVAILABLE = enum.auto()
+	# noinspection PyArgumentList
+	UPDATE = enum.auto()
+	# noinspection PyArgumentList
+	INCOMPATIBLE = enum.auto()
+	# noinspection PyArgumentList
+	DOWNLOADING = enum.auto()
+	# noinspection PyArgumentList
+	DOWNLOAD_FAILED = enum.auto()
+	# noinspection PyArgumentList
+	DOWNLOAD_SUCCESS = enum.auto()
+	# noinspection PyArgumentList
+	INSTALLING = enum.auto()
+	# noinspection PyArgumentList
+	INSTALL_FAILED = enum.auto()
+	# noinspection PyArgumentList
+	INSTALLED = enum.auto()  # installed, requires restart
+	# noinspection PyArgumentList
+	ENABLED = enum.auto()  # enabled after restart

```

Comment:
Is there a particular reason why DISABLED is missing? I assume that an update can be available for disabled add-ons. I think it should update but its state should remain disabled.

Reply:
We're doing this work in phases, currently the use-case being handled is browsing and installing new add-ons. To see disabled add-ons, you need to look at the add on manager dialog.

We plan to integrate all features into a single dialog. The next phase will be looking at updates.